apps/mark-scan/backend/src/custom-paper-handler/state_machine.ts

```diff
@@ -324,15 +369,87 @@ export function buildMachine(
           id: 'interpretScannedBallot',
           src: loadMetadataAndInterpretBallot,
           onDone: {
-            target: 'presenting_ballot',
+            target: 'transition_interpretation',
             actions: assign({
-              interpretation: (_, event) => {
-                return event.data;
-              },
+              interpretation: (_, event) => event.data,
             }),
           },
         },
       },
+      // Intermediate state to conditionally transition based on ballot interpretation
+      transition_interpretation: {
+        always: [
+          {
+            target: 'presenting_ballot',
+            cond: (context) =>
+              context.interpretation
+                ? context.interpretation[0].interpretation.type ===
+                  'InterpretedBmdPage'
+                : false,
+          },
+          {
+            target: 'blank_page_interpretation',
+            cond: (context) =>
+              context.interpretation
+                ? context.interpretation[0].interpretation.type === 'BlankPage'
+                : false,
+          },
+        ],
+      },
+      blank_page_interpretation: {
+        invoke: pollPaperStatus(),
+        entry: async (context) => {
+          await context.driver.presentPaper();
+        },
+        initial: 'presenting_paper',
+        // These nested states differ slightly from the top-level paper load states so we can't reuse the latter.
+        states: {
+          presenting_paper: {
+            // Heavier paper can fail to eject completely and trigger a jam state even though the paper isn't physically jammed.
+            // To work around this, we avoid ejecting to front. Instead, we present the paper so it's held by the device in a
+            // stable non-jam state. We instruct the poll worker to remove the paper directly from the 'presenting' state.
+            entry: async (context) => await context.driver.presentPaper(),
+            on: { NO_PAPER_ANYWHERE: 'accepting_paper' },
+          },
+          accepting_paper: {
+            on: {
+              PAPER_READY_TO_LOAD: 'load_paper',
+            },
+          },
+          load_paper: {
+            entry: async (context) => {
+              await context.driver.loadPaper();
+              await context.driver.parkPaper();
+            },
+            on: {
+              PAPER_PARKED: 'done',
+              NO_PAPER_ANYWHERE: 'accepting_paper',
+            },
+          },
+          done: {
```

Comment:
Could probably remove this state by adding the `assign` action contained here to the transition from `load_paper`

Reply:
Agree the `assign` call can go anywhere, but the state machine breaks when I get rid of the `done` state and transitioning from `load_paper`: https://app.circleci.com/pipelines/github/votingworks/vxsuite/12078/workflows/b8f38ca0-e4cd-4e24-bc63-48b103b367a0/jobs/400580

It's expecting the target state to be a sibling of `load_paper` ie. a child of `blank_page_interpretation` _unless_ we transition from the `done` state. When transitioning from `done` it can successfully transition to top-level states.